samples/pyarrow/append_rows_with_arrow.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+import decimal
+import pandas as pd
+import pyarrow as pa
+
+from google.cloud import bigquery
+from google.cloud.bigquery import enums
+from google.cloud.bigquery_storage_v1 import types as gapic_types
+from google.cloud.bigquery_storage_v1.writer import AppendRowsStream
+
+
+def bqstorage_write_client():
+    from google.cloud import bigquery_storage_v1
+
+    return bigquery_storage_v1.BigQueryWriteClient()
+
+
+def make_table(project_id, dataset_id, bq_client):
+    schema = [
+        bigquery.SchemaField("bool_col", enums.SqlTypeNames.BOOLEAN),
+        bigquery.SchemaField("int64_col", enums.SqlTypeNames.INT64),
+        bigquery.SchemaField("float64_col", enums.SqlTypeNames.FLOAT64),
+        bigquery.SchemaField("numeric_col", enums.SqlTypeNames.NUMERIC),
+        bigquery.SchemaField("bignumeric_col", enums.SqlTypeNames.BIGNUMERIC),
+        bigquery.SchemaField("string_col", enums.SqlTypeNames.STRING),
+        bigquery.SchemaField("bytes_col", enums.SqlTypeNames.BYTES),
+        bigquery.SchemaField("date_col", enums.SqlTypeNames.DATE),
+        bigquery.SchemaField("datetime_col", enums.SqlTypeNames.DATETIME),
+        bigquery.SchemaField("time_col", enums.SqlTypeNames.TIME),
+        bigquery.SchemaField("timestamp_col", enums.SqlTypeNames.TIMESTAMP),
+        bigquery.SchemaField("geography_col", enums.SqlTypeNames.GEOGRAPHY),
+        bigquery.SchemaField(
+            "range_date_col", enums.SqlTypeNames.RANGE, range_element_type="DATE"
+        ),
+        bigquery.SchemaField(
+            "range_datetime_col",
+            enums.SqlTypeNames.RANGE,
+            range_element_type="DATETIME",
+        ),
+        bigquery.SchemaField(
+            "range_timestamp_col",
+            enums.SqlTypeNames.RANGE,
+            range_element_type="TIMESTAMP",
+        ),
+    ]
+    table_id = "append_rows_w_arrow_test"
+    table_id_full = f"{project_id}.{dataset_id}.{table_id}"
+    bq_table = bigquery.Table(table_id_full, schema=schema)
+    created_table = bq_client.create_table(bq_table)
+
+    return created_table
+
+
+def create_stream(bqstorage_write_client, table):
+    stream_name = f"projects/{table.project}/datasets/{table.dataset_id}/tables/{table.table_id}/_default"
+    request_template = gapic_types.AppendRowsRequest()
```

Comment:
The structure here looks a little off to me.  I'd expect this template request to have the arrow schema, but you're setting it in the data request.

Reply:
No, schema evolution is a more complex topic, let's focus on streaming data for a consistent schema here.